cards.js

```diff
@@ -17,7 +17,27 @@ function shuffle(array){
 }
 
 function buildDeck(){
+  var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
+  var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
+  var deck = []
+  for(v in values){
+    for(s in suits){
```

Comment:
This works! Typically, `for in` loops are used for looping through attributes of objects, and the `for( var i =0; i < length; i++ )` loops are used for looping through arrays.

I had to google this but found a helpful explanation why here - http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-such-a-bad-idea